### PR TITLE
refactor(eg/channel): deprecate the channel_id filter and refactor the resource codes

### PR DESCRIPTION
**Pull request by Lance52259**
**What this PR does / why we need it**:
The API document has offline the channel_id parameter, so, we mark it as deprecated and remove it from the markdown file definition.

**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:

**Release note**:

```release-note
1. deprecate the channel_id filter
```

## PR Checklist

* [x] Tests added/passed.

```
make testacc TEST='./huaweicloud/services/acceptance/eg' TESTARGS='-run=TestAccDataCustomEventChannels_filterById'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/eg -v -run=TestAccDataCustomEventChannels_filterById -timeout 360m -parallel 4
=== RUN   TestAccDataCustomEventChannels_filterById
=== PAUSE TestAccDataCustomEventChannels_filterById
=== CONT  TestAccDataCustomEventChannels_filterById
--- PASS: TestAccDataCustomEventChannels_filterById (37.01s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/eg        37.113s
```

* [x] Documentation updated.
* [ ] Schema updated.
* [ ] CheckDeleted.

  - **a. During query operation (Read Context)**
    aa. Resource not found
    \>>>>>> Paste the screenshot here <<<<<<

    
  - **b. During delete/disassociate/unbind operation (Delete Context)**
    ba. Resource not found
    \>>>>>> Paste the screenshot here <<<<<<

### CODE REVIEW

**Comment by Lance52259** [comment]
make testacc TEST='./huaweicloud/services/acceptance/eg' TESTARGS='-run=TestAccDataCustomEventChannels_'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/eg -v -run=TestAccDataCustomEventChannels_ -timeout 360m -parallel 4
=== RUN   TestAccDataCustomEventChannels_basic
=== PAUSE TestAccDataCustomEventChannels_basic
=== RUN   TestAccDataCustomEventChannels_filterById
=== PAUSE TestAccDataCustomEventChannels_filterById
=== RUN   TestAccDataCustomEventChannels_filterByEpsId
=== PAUSE TestAccDataCustomEventChannels_filterByEpsId
=== CONT  TestAccDataCustomEventChannels_basic
=== CONT  TestAccDataCustomEventChannels_filterByEpsId
=== CONT  TestAccDataCustomEventChannels_filterById
--- PASS: TestAccDataCustomEventChannels_filterByEpsId (51.88s)
--- PASS: TestAccDataCustomEventChannels_filterById (59.43s)
--- PASS: TestAccDataCustomEventChannels_basic (61.23s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/eg        61.301s

File: huaweicloud/services/eg/data_source_huaweicloud_eg_custom_event_channels.go [hunk 1]
```diff
@@ -98,35 +104,80 @@ func DataSourceCustomEventChannels() *schema.Resource {
...
+			"updated_at":            utils.PathSearch("updated_time", channel, nil),
+		})
+	}
+
+	return result
+}
+
+func filterEventChannels(cfg *config.Config, d *schema.ResourceData, channels []interface{}) []interface{} {
+	result := make([]interface{}, len(channels))
+	copy(result, channels)
```
Comment (deer-hang) [inline huaweicloud/services/eg/data_source_huaweicloud_eg_custom_event_channels.go]: Seems like this line is unnecessary.

**Comment by Lance52259** [comment]
image

**Comment by Lance52259** [comment]
```
../coverage.sh -o eg -f TestAccDataCustomEventChannels
Prepare to calculate the coverage the following command:
TF_ACC=1 go test "./huaweicloud/services/acceptance/eg" -v -coverprofile="./huaweicloud/services/acceptance/eg_coverage.cov" -coverpkg="./huaweicloud/services/eg" -run TestAccDataCustomEventChannels -timeout 360m -parallel 10
=== RUN   TestAccDataCustomEventChannels_basic
=== PAUSE TestAccDataCustomEventChannels_basic
=== RUN   TestAccDataCustomEventChannels_filterById
=== PAUSE TestAccDataCustomEventChannels_filterById
=== RUN   TestAccDataCustomEventChannels_filterByEpsId
=== PAUSE TestAccDataCustomEventChannels_filterByEpsId
=== CONT  TestAccDataCustomEventChannels_basic
=== CONT  TestAccDataCustomEventChannels_filterByEpsId
=== CONT  TestAccDataCustomEventChannels_filterById
--- PASS: TestAccDataCustomEventChannels_filterByEpsId (53.64s)
--- PASS: TestAccDataCustomEventChannels_filterById (58.65s)
--- PASS: TestAccDataCustomEventChannels_basic (60.52s)
PASS
coverage: 13.1% of statements in ./huaweicloud/services/eg
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/eg        60.585s coverage: 13.1% of statements in ./huaweicloud/services/eg
```

File: huaweicloud/services/eg/data_source_huaweicloud_eg_custom_event_channels.go (hunk 1 above)
Comment (Lance52259) [reply huaweicloud/services/eg/data_source_huaweicloud_eg_custom_event_channels.go]: already removed.

**Pull request merged**